### PR TITLE
I1339 step3 cleanup test

### LINKED ISSUE #1339: Remove Scala SmvDataSet interface classes
Keep the `SmvDataSet` base class and all necessary classes for Python side modules continue to work as is. Remove every other classes derived from `SmvDataSet` on Scala side.

### PR DESCRIPTION
**Pull request by ninjapapa**
Fixed #1339 

Removed the following from Scala side:
* `Using[SmvRunConfig]`
* `SmvAncillary`/`SmvHierarchy` 
* `SmvJdbcTable`
* scala shell commands
* all scala integration tests

### CODE REVIEW

File: src/main/scala/org/tresamigos/smv/SmvDataSet.scala [hunk 1]
```diff
@@ -710,42 +694,6 @@ private[smv] abstract class SmvInputDataSet extends SmvDataSet {
   def run(df: DataFrame) = df
 }
 
-/**
- * Wrapper for a database table accessed via JDBC
- */
-class SmvJdbcTable(override val tableName: String)
```
Comment (AliTajeldin) [inline src/main/scala/org/tresamigos/smv/SmvDataSet.scala]: Are we dropping support for jdbc input?  Should we wait to remove this after we have the python equivalent?

Comment (laneb) [reply src/main/scala/org/tresamigos/smv/SmvDataSet.scala]: We have a Python implementation based on the DataFrameReader. Should be ready to drop the Scala implementation.

Comment (ninjapapa) [reply src/main/scala/org/tresamigos/smv/SmvDataSet.scala]: https://github.com/TresAmigosSD/SMV/blob/i1339_step3_cleanup_test/src/main/python/smv/smvinput.py#L404

Comment (AliTajeldin) [reply src/main/scala/org/tresamigos/smv/SmvDataSet.scala]: cool beans.

**Pull request merged**